samples: driver: memc: Fix minor issues

dnltz opened the pull request:
Fix the code-block inside the README and the missing include for memcpy. Make the sample more verbose and show if it's still running by printing something after each compare step.

dnltz committed:
samples: drivers: memc: Fix README

The output example has not the correct indentation for the code-block
directive and therefore adds this output as normal text.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
samples: drivers: memc: Add missing include

Add string.h because memcpy requires this include file.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
samples: drivers: memc: Make sample more verbose

During a memory mismatch, the sample only prints an error message.
Also print the read and write buffer to see the actual difference.

Moreover, add a print for each compare step to see the sample is still
running.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

kartben commented on samples/drivers/memc/src/main.c:
```diff
@@ -69,8 +70,11 @@ int main(void)
+			dump_memory(memc_read_buffer, BUF_SIZE);
 			return 0;
 		}
+		printk("Check (%i/%i) passed!\n", j, MEMC_SIZE / BUF_SIZE);
```
looks like you need to update the sample output in the README then?

dnltz replied:
yes, thanks. I also fixed the `Writing to memory region with...` printk because MEMC_BASE is a long long int but %p is used.

dnltz committed:
samples: driver: memc: Fix print argument

The sample wants to print the memory controller base address. However,
MEMC_BASE is used as argument in printf which is defined as long long
int. Use the memc pointer instead to print the base address.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

kartben commented on samples/drivers/memc/README.rst:
```diff
@@ -35,14 +35,20 @@ Sample Output
+  Check (8191/8191) passed!
+
+  First 1KB of Data in memory:
+  \=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=\=
```
```suggestion
  ===========================
```

kartben replied:
not quite sure why this was ever escaped :)

kartben reviewed (changes requested):
thanks! just spotted one last weird escaped string leftover that doesn't make sense in this context (and breaks the [rendered doc](https://builds.zephyrproject.io/zephyr/pr/69496/docs/samples/drivers/memc/README.html#sample-output))

kartben commented on samples/drivers/memc/README.rst:
```diff
@@ -35,14 +35,20 @@ Sample Output
-....
-
-Read data matches written data
+  *** Booting Zephyr OS build zephyr-v3.2.0-2686-gd52d828c2bdc ***
```
nit: might not be absurd to update to the 3.6.99 version you ran this on, since it's not really the 3.2 sample anymore.

dnltz replied:
good point. Update to `v3.6.0-25`
